File: flutter/storyly_placement_flutter/lib/storyly_log.dart
import 'storyly_placement_platform_interface.dart';

/// Log levels for Storyly Placement. [off] is the default.
enum STRLogLevel {
  debug,
  warning,
  error,
  off,
}

/// Log level control for the placement SDK.
///
/// Logs are off by default. Setting a level applies to both the
/// cross-platform wrapper and the native Storyly Placement SDK.
class StorylyLog {
  StorylyLog._();

  /// Sets the minimum log level for Storyly Placement.
  /// [STRLogLevel.debug] is the most verbose, [STRLogLevel.off] (default)
  /// silences all output.
  static Future<void> setLogLevel(STRLogLevel level) {
    return StorylyPlacementFlutterPlatform.instance.setLogLevel(level.name);
  }
}

File: react-native/storyly-placement-react-native/src/StorylyLog.ts
/**
 * Storyly Log - log level control for the placement SDK.
 *
 * Logs are off by default. Setting a level applies to both the
 * cross-platform wrapper and the native Storyly Placement SDK.
 */

import StorylyPlacementProviderNative from './native/StorylyPlacementProviderNative';

export type STRLogLevel = 'debug' | 'warning' | 'error' | 'off';

export const StorylyLog = {
  /**
   * Sets the minimum log level for Storyly Placement.
   * `debug` is the most verbose, `off` (default) silences all output.
   */
  setLogLevel(level: STRLogLevel): void {
    StorylyPlacementProviderNative.setLogLevel(level);
  },
};
